app/controllers/listings_controller.rb

```diff
@@ -1,5 +1,14 @@
 class ListingsController < ApplicationController
   def index
-    @listings = Listing.all
+    @listings = Listing.where(nil)
+    filtering_params(params).each do |key, value|
+      @listings = @listings.public_send(key, value) if value.present?
```

Comment:
Also, I totally just copied this from this dude's blog: http://www.justinweiss.com/blog/2014/02/17/search-and-filter-rails-models-without-bloating-your-controller/